2009-flu-uk.qmd

```diff
@@ -1 +1,437 @@
+- $h(t)$, the reduction of contact due to holidays, assumed piecewise constant,
+- $\rho$, the ascertainment probability i.e. the proportion of infections we expect to ascert as cases,
+- $\eta$, the size parameter of our Negative Binomial observation model,
+- $N$ the total size of the population in England and Wales in 2009.
+
+### Odin code of the model
+
+```{r}
+library(odin2)
+library(dust2)
+library(monty)
+```
+
+Below is the `odin` code for our model - the structure matches closely the mathematical formulation we derived in the previous section. Note the `Cases <- data()` lines telling that `Cases` are observations though the model will compile and work even if not linked with any dataset. However providing data is essential to compute a likelihood with `dust` or derive a `monty` likelihood statistical model of our system.
+
+The compilation step takes time. The model needs to be first transpiled (e.g. translated from one programming language to another) in order to create some C++ code and then that code is compiled and loaded in the R environnment. After this, a "fast" model can be called from the R environment and used to simulate scenarios or perform inference. 
+
+```{r, echo = TRUE}
+seir <- odin2::odin({
+# initial conditions
+initial(S) <- (1 - 2 * alpha) * N
+initial(E) <- alpha * N
+initial(I) <- alpha * N
+initial(R) <- 0
+initial(incidence, zero_every = 7) <- 0
+
+# equations
+deriv(S) <- - hol * beta * S * I / N
+deriv(E) <- hol * beta * S * I / N - gamma * E
+deriv(I) <- gamma * E - sigma * I
+deriv(R) <- sigma * I
+deriv(incidence) <- gamma * E
+
+# parameter values
+R_0 <- parameter(1.5)
+L <- 1
+D <- 1.25
+alpha <- parameter(1e-4) # initial proportion
+N <- 55000000
+
+# convert parameters
+hol <- interpolate(h_times, h_values, "constant")
+h_times <- parameter()
+h_values <- parameter()
+dim(h_times) <- parameter(rank=1)
+dim(h_values) <- length(h_times)
+gamma <- 1 / L
+sigma <- 1 / D
+beta <- R_0 * sigma
+
+# observation model
+rho <- parameter(0.1)
+eta <- parameter(10)
+Cases <- data()
+Cases ~ NegativeBinomial(size = eta, mu = rho*incidence)
+})
+```
+
+### Running the model
+
+Once the model is compiled, it is possible to generate one (or more!) instance of your model by using the `dust_system_create()` and your model generator.
+
+```{r}
+mod <- dust_system_create(seir,
+                          pars = list(h_times = hol_t, h_values = hol_v))
+```
+
+```{r}
+# A vector with parameter values
+# First parameter -> proportion of population initially infected
+# Second argument -> R_0
+# Third argument -> Proportion of infection detected 
+par0 <- c(5e-5, 1.3, 0.1, 10)
+
+# This vector should be passed as a list
+dust_system_update_pars(sys = mod,
+                        pars = list(alpha=par0[1],
+                                    R_0=par0[2],
+                                    h_times = hol_t,
+                                    h_values= hol_v))
```

Comment:
```suggestion
                        pars = list(alpha = par0[1],
                                    R_0 = par0[2],
                                    h_times = hol_t,
                                    h_values = hol_v))
```

Reply:
I would probably write
`pars <- list(...)`
first and then just use `pars` as an input here and anywhere else

Reply:
Also, probably include `rho` and `N` in the list

Reply:
Change done.